src/uct/ib/dc/dc_mlx5.c

```diff
@@ -409,6 +410,12 @@ static ucs_status_t uct_dc_mlx5_iface_create_dci(uct_dc_mlx5_iface_t *iface,
     uct_rc_txqp_cleanup(&iface->super.super, &dci->txqp);
 err_qp:
     uct_ib_mlx5_destroy_qp(md, &dci->txwq.super);
+#if HAVE_DC_DV
```

Comment:
I think better remove uct_ib_mlx5_iface_get_res_domain from uct_ib_mlx5_iface_fill_attr and call it separately. also, make uct_ib_mlx5_iface_fill_attr return void.
reason: all "fill_attr" functions should not have side effects. it's better the caller function would use uct_ib_mlx5_iface_get_res_domain explicitly so it would be clear what needs to be cleaned-up in case of error.

Reply:
@yosefe I've refined the implementation of ```uct_ib_mlx5_iface_fill_attr``` and release ```resource domain``` in the proper way. PR has passed CI test cases.

Reply:
@yosefe @Artemy-Mellanox 
What's the next step for this PR?